### PR TITLE
Bug 1127261 - stop discovery before pair with a device, r=arthur

### PR DESCRIPTION
**Pull request by ian-liu**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[ian-liu Ian Liu (ian-liu)](https://github.com/ian-liu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3a7abd82b7ae)

File: apps/settings/js/modules/bluetooth/bluetooth_context.js [hunk 1]
```diff
@@ -797,12 +802,20 @@ define(function(require) {
...
+      return this.stopDiscovery().then(() => {
+        return this._defaultAdapter.pair(address).then(() => {
+          Debug('pair(): Resolved with void value');
+        }, (reason) => {
+          Debug('pair(): Reject with this reason: ' + reason);
+          return Promise.reject(reason);
+        });  
       }, (reason) => {
-        Debug('pair(): Reject with this reason: ' + reason);
+        Debug('pair(): Reject in stopDiscovery with this reason: ' + reason);
```
Comment (crh0716) [inline apps/settings/js/modules/bluetooth/bluetooth_context.js]: This is not required as in `this.stopDiscovery` you already printed the error message.

File: apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js [hunk 2]
```diff
@@ -1165,13 +1189,45 @@ suite('BluetoothContext', function() {
       test('_defaultAdapter.pair() should be called with addrss ',
```
Comment (crh0716) [inline apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js]: type: address

**Comment by try-server-hook** [comment]
[ian-liu Ian Liu (ian-liu)](https://github.com/ian-liu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7aac00c07d5b)

**Comment by ian-liu** [comment]
Passed on try-server. We can merge the pull request now.

**Pull request merged**